packages/experiment-tag/src/experiment.ts

```diff
@@ -4,461 +4,641 @@ import {
+    this.experimentClient.setUser(user);
+
+    if (this.config.useDefaultNavigationHandler) {
+      this.setDefaultNavigationHandler([
+        ...this.localFlagKeys,
+        ...this.remoteFlagKeys,
+      ]);
+    }
 
-  // initialize the experiment
-  globalScope.webExperiment = Experiment.initialize(apiKey, {
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    // @ts-ignore
-    internalInstanceNameSuffix: 'web',
-    initialFlags: initialFlags,
-    // timeout for fetching remote flags
-    fetchTimeoutMillis: 1000,
-    pollOnStart: false,
-    fetchOnStart: false,
-    ...config,
-  });
+    // apply local variants
+    this.applyVariants({ flagKeys: this.localFlagKeys });
 
-  // evaluate variants for page targeting
-  const variants: Variants = globalScope.webExperiment.all();
+    if (!this.isRemoteBlocking) {
+      // Remove anti-flicker css if remote flags are not blocking
+      this.globalScope.document.getElementById?.('amp-exp-css')?.remove();
+    }
 
-  for (const [key, variant] of Object.entries(variants)) {
-    // only apply antiflicker for remote flags active on the page
-    if (
-      remoteFlagKeys.has(key) &&
-      variant.metadata?.blockingEvaluation &&
-      variant.metadata?.segmentName !== 'Page not targeted' &&
-      variant.metadata?.segmentName !== 'Page is excluded'
-    ) {
-      isRemoteBlocking = true;
-      // Apply anti-flicker CSS to prevent UI flicker
-      applyAntiFlickerCss();
+    if (this.remoteFlagKeys.length === 0) {
+      return;
     }
+
+    await this.fetchRemoteFlags();
+    // apply remote variants - if fetch is unsuccessful, fallback order: 1. localStorage flags, 2. initial flags
+    this.applyVariants({ flagKeys: this.remoteFlagKeys });
   }
 
-  // If no integration has been set, use an amplitude integration.
-  if (!globalScope.experimentIntegration) {
-    const connector = AnalyticsConnector.getInstance('$default_instance');
-    globalScope.experimentIntegration = new AmplitudeIntegrationPlugin(
+  /**
+   * Get singleton of the {@link DefaultWebExperimentClient} if it has already been initialized.
+   * If not, initialize the client and return the instance.
+   * @param apiKey
+   * @param initialFlags
+   * @param config
+   */
+  static async getInstance(
+    apiKey: string,
+    initialFlags: string,
+    config: WebExperimentConfig = {},
+  ): Promise<DefaultWebExperimentClient> {
+    const globalScope = getGlobalScope();
+    if (!globalScope) {
+      throw new Error(
+        'Amplitude Web Experiment Client could not be initialized.',
+      );
+    }
+    // if the client has already been initialized, return the existing instance
+    if (globalScope?.webExperiment) {
+      return globalScope.webExperiment;
+    }
+    const webExperiment = new DefaultWebExperimentClient(
       apiKey,
-      connector,
-      0,
+      initialFlags,
+      config,
     );
+    await webExperiment.start();
```

Comment:
pull the call to `start` out of `getInstance` and make the function synchronous then the caller would need to call start as needed.

also make `start` idempotent (can only be called once per instance).